From f818d048ac93070a808d2e78f36e323949b65194 Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Thu, 22 Jun 2017 15:04:01 -0400
Subject: [PATCH 2434/2500] drm/amdgpu: Fix non-contig ptrace VRAM access
 crossing nodes

Fix ptrace VRAM access when crossing non-contiguous VRAM node
boundaries.

Change-Id: I9c876145d0087a78fc17f0a76b9ac47f55a4f35b
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
index 18f2c9b..bfc2265 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
@@ -1554,6 +1554,10 @@ static int amdgpu_ttm_bo_access_vram(struct amdgpu_bo *abo,
 		buf = (uint8_t *)buf + bytes;
 		pos += bytes;
 		len -= bytes;
+		if (pos >= (nodes->start + nodes->size) << PAGE_SHIFT) {
+			++nodes;
+			pos = (nodes->start << PAGE_SHIFT);
+		}
 	}
 
 	return result;
-- 
2.7.4

